R/as_list.R

```diff
@@ -55,8 +58,13 @@ as_list.xml_node <- function(x, ns = character(), ...) {
 
   # Add xml attributes as R attributes
   attr <- xml_attrs(x, ns = ns)
-  if (length(attr) > 0)
-    attributes(out) <- as.list(attr)
+  if (length(attr) > 0) {
+    # escape special names
+    special <- names(attr) %in% c("class", "comment", "dim", "dimnames", "names", "row.names", "tsp")
+    if(any(special))
```

Comment:
This conditional is unnecessary, the assignment will work without error regardless of the value of special

``` r
letters[logical(0)] <- 1
letters[FALSE] <- 1
letters[NULL] <- 1
letters
#>  [1] "a" "b" "c" "d" "e" "f" "g" "h" "i" "j" "k" "l" "m" "n" "o" "p" "q"
#> [18] "r" "s" "t" "u" "v" "w" "x" "y" "z"
```